<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Project MAR1</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabContainer">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="mainTab">
       <attribute name="title">
        <string>Main</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="spacing">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="cycleLabel">
            <property name="text">
             <string>Cycle</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTableWidget" name="cycle">
            <row>
             <property name="text">
              <string>ADD R0, R1</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>SUB R0, R2</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>STORE R0, R4</string>
             </property>
            </row>
            <column>
             <property name="text">
              <string>1</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>2</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>3</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>4</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>5</string>
             </property>
            </column>
            <item row="0" column="0">
             <property name="text">
              <string>F</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>200</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="0" column="1">
             <property name="text">
              <string>D</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>200</red>
                <green>255</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="0" column="2">
             <property name="text">
              <string>E</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>200</red>
                <green>200</green>
                <blue>255</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="1" column="0">
             <property name="text">
              <string>S</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>255</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
             <property name="foreground">
              <brush brushstyle="NoBrush">
               <color alpha="255">
                <red>0</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="1" column="1">
             <property name="text">
              <string>F</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>200</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="1" column="2">
             <property name="text">
              <string>D</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>200</red>
                <green>255</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="1" column="3">
             <property name="text">
              <string>E</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>200</red>
                <green>200</green>
                <blue>255</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="2" column="0">
             <property name="text">
              <string>S</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>255</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="2" column="1">
             <property name="text">
              <string>S</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>255</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="2" column="2">
             <property name="text">
              <string>F</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>200</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="2" column="3">
             <property name="text">
              <string>D</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>200</red>
                <green>255</green>
                <blue>200</blue>
               </color>
              </brush>
             </property>
            </item>
            <item row="2" column="4">
             <property name="text">
              <string>E</string>
             </property>
             <property name="background">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>200</red>
                <green>200</green>
                <blue>255</blue>
               </color>
              </brush>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="pipelineLabel">
            <property name="text">
             <string>Pipeline</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGraphicsView" name="pipeline"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="assemblyLabel">
            <property name="text">
             <string>Assembly Code</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="assemblyCode">
            <property name="font">
             <font>
              <family>Monospace</family>
             </font>
            </property>
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Monospace'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;STORE R0 1&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;STORE R1 2&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;ADD R0 R1&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="executeButton">
            <property name="text">
             <string>Execute</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="nextStateButton">
            <property name="text">
             <string>Next State</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="memoryTab">
       <attribute name="title">
        <string>Memory</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="registerLabel">
            <property name="text">
             <string>Register Table</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTableView" name="registerTable">
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarAlwaysOff</enum>
            </property>
            <attribute name="horizontalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="memoryLabel">
            <property name="text">
             <string>Physical Memory</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTableView" name="memoryTable">
            <attribute name="horizontalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuExecution">
    <property name="title">
     <string>E&amp;xecution</string>
    </property>
    <addaction name="actionExecute"/>
    <addaction name="actionNext_State"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuExecution"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionExecute">
   <property name="text">
    <string>Execute</string>
   </property>
  </action>
  <action name="actionNext_State">
   <property name="text">
    <string>Next State</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionExecute</sender>
   <signal>triggered()</signal>
   <receiver>executeButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>164</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>761</x>
     <y>697</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNext_State</sender>
   <signal>triggered()</signal>
   <receiver>nextStateButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>761</x>
     <y>733</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
